# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerScope.cpp
# Commit: d801fdeac90b
# Full Hash: d801fdeac90bc63de7f7efd090ced48bf871121a
# Author: Tom Ritter <tom@mozilla.com>
# Date: 2022-11-29 08:40:32
# Regressor Bug: 1778510
# File Overlap Count: 1
# Description:
#   Bug 1778510: Add RTPCallerType to GlobalObject r=asuth
#   
#   This centralizes the logic in one place.
#   
#   In order to do this, we will need to check the principal
# ==============================================================================

diff -r 5af032c628c0 -r d801fdeac90b dom/workers/WorkerScope.cpp
--- a/dom/workers/WorkerScope.cpp	Mon Nov 28 18:04:06 2022 +0000
+++ b/dom/workers/WorkerScope.cpp	Mon Nov 28 18:04:07 2022 +0000
@@ -276,6 +276,10 @@
   return mShouldResistFingerprinting;
 }
 
+bool WorkerGlobalScopeBase::IsSystemPrincipal() const {
+  return mWorkerPrivate->UsesSystemPrincipal();
+}
+
 uint32_t WorkerGlobalScopeBase::GetPrincipalHashValue() const {
   AssertIsOnWorkerThread();
   return mWorkerPrivate->GetPrincipalHashValue();